geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala

```diff
@@ -145,28 +156,78 @@ class StreamFeatureStore(entry: ContentEntry,
 }
 
 object StreamDataStoreParams {
-  val StreamDatastoreConfig = new GeoMesaParam[String]("geomesa.stream.datastore.config", "", optional = false)
-  val CacheTimeout = new GeoMesaParam[Integer]("geomesa.stream.datastore.cache.timeout", "", optional = false, default = 10)
+  // TODO: this assumes user is using a GenericSimpleFeatureSource and an associated Camel context
+  lazy val ctx: CamelContext = {
+    val context = new DefaultCamelContext()
+    context.start()
+    context
+  }
+
+  val StreamDatastoreConfig = new GeoMesaParam[String]("geomesa.stream.datastore.config", "", optional = true, largeText = true)
+  val CamelRouteConfig = new GeoMesaParam[String]("geomesa.stream.datastore.config.route", "The camel route", optional = true, largeText = true)
+  val SftConfig = new GeoMesaParam[String]("geomesa.stream.datastore.config.sft", "Name of the SFT", optional = true)
+  val ConverterConfig = new GeoMesaParam[String]("geomesa.stream.datastore.config.converter", "Name of the converter", optional = true)
+  val ThreadsConfig = new GeoMesaParam[Integer]("geomesa.stream.datastore.config.num.threads", "Number of threads to use", optional = true, default = 1)
+  val CacheTimeout = new GeoMesaParam[Integer]("geomesa.stream.datastore.cache.timeout", "", optional = false, default = -1)
+  val NamespaceParam     = new GeoMesaParam[String]("namespace", "Namespace")
+}
+
+object StreamDataStoreFactory {
+  val stores: ju.Map[String, StreamDataStore] = Collections.synchronizedMap(Maps.newHashMap[String, StreamDataStore]())
 }
 
 class StreamDataStoreFactory extends DataStoreFactorySpi {
 
   import StreamDataStoreParams._
+  private val log = LoggerFactory.getLogger(classOf[StreamDataStoreFactory])
 
   override def createDataStore(params: ju.Map[String, java.io.Serializable]): DataStore = {
-    val confString = StreamDatastoreConfig.lookup(params)
+    // TODO: note, if user is requesting a SDS via a inline config, it'll get a default name and namespace of ":"
```

Comment:
@elahrvivaz do we have a standard way of handling this?

Reply:
we have a standard way of handling datastore namespaces, but i think the stream store is actually using the namespace, which we don't do anywhere else.